src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -346,6 +346,35 @@ def _visit_neighbor_sum(self, node: foast.Call, **kwargs):
             type=return_type,
         )
 
+    def _visit_broadcast(self, node: foast.Call, **kwargs):
+        field_type = cast(ct.FieldType, node.args[0].type)
+        broadcast_dims = [cast(ct.DimensionType, elt.type).dim for elt in node.args[1].elts]
+
+        if not set(field_type.dims).issubset(set(broadcast_dims)):
+            raise FieldOperatorTypeDeductionError.from_foast_node(
+                node,
+                msg=f"Incompatible broadcast dimensions in {node.func.id}. Expected "
+                f"broadcast dimension is missing {set(field_type.dims).difference(set(broadcast_dims))}",
+            )
+
+        for bdim in broadcast_dims:
+            if bdim in field_type.dims:
+                continue
+            field_type.dims.append(bdim)
+
+        return_type = ct.FieldType(
+            dims=field_type.dims,
+            dtype=field_type.dtype,
+        )
```

Comment:
```suggestion
        return_type = ct.FieldType(
            dims=broadcast_dims,
            dtype=field_type.dtype,
        )
```
I don't see that the loop is needed. After it `set(field_type.dims) == set(broadcast_dims)`, but you might have altered the ordering of `broadcast_dims` which is not necessary, but actually harmful. For example `broadcast(some_j_field, (I,J,K))` for `some_j_field: Field[[J], dtype]` will give you `Field[[J, I, K], dtype]` currently. Additionally modifying `field_type.dims` is wrong as you also change `node.args[0].type` by that.

Reply:
Make sense, thanks for the suggestion. Removed the for loop now.